modules/ogg/ogg_packet_sequence.cpp

```diff
@@ -175,6 +175,9 @@ uint32_t OGGPacketSequencePlayback::seek_page_internal(int64_t granule, uint32_t
 			break;
 		}
 	}
+	if (bisection_page == before_page_inclusive && granule <= (int64_t)ogg_packet_sequence->page_granule_positions[bisection_page]) {
```

Comment:
I don't think this is the right solution. It seems like any time that `bisection_page == before_page_inclusive` automatically becomes a failure case because the method is called again with `bisection_page + 1` as `after_page_inclusive` which means the base case of `before_page_inclusive == after_page_inclusive` is never triggered because in that scope, `before_page_inclusive` is greater than `after_page_inclusive`. Satisfying that constraint is impossible. Could we try making the base case `before_page_inclusive >= after_page_inclusive` instead of using `==`?

If we do that though, it's not immediately clear to me whether we should return `before_page_inclusive` or `after_page_inclusive` in the base case. It could cause problems if we return the wrong one, because it could possibly be an empty page. Although, an empty page isn't the end of the world because playback should handle those gracefully.

I'm really sorry for how confusing this code is. My head is spinning trying to understand it now, months later.

Reply:
Another option is modifying the loop to maybe be `test_page < before_page_inclusive` instead of `test_page <= before_page_inclusive`? I haven't tested either of these ideas but they seem like where I'd start. Thank you so much for figuring out where the problem is, by the way.

Reply:
In the MRP, it's kind of like searching for 0 inside range `[0, 2]`:

| idx | pos | size |
|------|------|-------|
| 0     | 0     | 1     |
| 1     | 0     | 0     |
| 2     | 0     | 1     |

In this case, `bisection_page` first picks 2, and recurreset it to -1, and then searches forward and picks 0.